test/libsolidity/semanticTests/immutable/getter_call_in_constructor.sol

```diff
@@ -0,0 +1,15 @@
+contract A {
+    uint immutable public x = 1;
+    uint public y;
+    constructor() public {
+        y = this.x();
+    }
+}
+contract C {
+    function f() public returns (bool) {
+        try new A() { return false; }
+        catch { return true; }
```

Comment:
What a way to test this :-)!

Reply:
Interestingly, apparently the test case fails with a revert, though...

Reply:
Of course, because of the extcodesize check

Reply:
Ah sorry. Hm. need to check.

Reply:
I meant apparently ``C.f`` reverts and not only ``new A()``... that shouldn't happen, should it?

Reply:
It fails on homestead but succeeds starting from tangerineWhistle.

Reply:
Ah right - I guess there is not enough gas left to do the returning